### PR TITLE
add jsonSpecCID to createQuestAndQueue

### PR DESCRIPTION
**Pull request by waynehoover**

### CODE REVIEW

**Review by jimobrien — APPROVED** [review]

File: contracts/Quest.sol [hunk 1]
```diff
@@ -66,6 +67,15 @@ contract Quest is ReentrancyGuardUpgradeable, PausableUpgradeable, OwnableUpgrad
         __ReentrancyGuard_init();
     }
 
+    /// @dev set jsonSpecCID only if its empty
+    /// @param jsonSpecCID_ The jsonSpecCID to set
+    function setJsonSpecCID(string memory jsonSpecCID_) external onlyOwner {
+        require(bytes(jsonSpecCID_).length > 0, 'jsonSpecCID cannot be empty');
+        require(bytes(jsonSpecCID).length == 0, 'jsonSpecCID already set');
```
Comment (jonathandiep) [inline contracts/Quest.sol:74]: So once set, we cannot edit?

Comment (jonathandiep) [reply contracts/Quest.sol:74]: I guess I kinda agree with this. So owner can't change requirements in middle of quest

Wondering if we can allow edits until the quest starts?

File: contracts/Quest.sol [hunk 2]
```diff
@@ -66,6 +67,15 @@ contract Quest is ReentrancyGuardUpgradeable, PausableUpgradeable, OwnableUpgrad
...
+
+        jsonSpecCID = jsonSpecCID_;
```
Comment (jonathandiep) [inline contracts/Quest.sol:76]: Should we emit event?

**Review by jonathandiep — APPROVED** [review]
Left a couple comments, but lgtm!

**Pull request merged**